### ISSUE TITLE
Command output redirect generates "empty" files

### ISSUE THREAD

**Issue by kartben** [dottspina/dtsh#7]
When doing e.g. the following command (which works fine without redirecting output):

```
ls -R --format NcrYd > board.html
```

I get an "empty" file (except for the template). Same happens with .txt and .svg

```html
<!DOCTYPE html>
<html>

<head>
<meta charset="UTF-8">

<style>

body {
    color: #000000;
    background-color: #ffffff;
}
</style>
</head>

<body>

    <pre style="font-family:Courier New, monospace"><code></code></pre>

</body>

</html>
```

This is with v0.2.3

**Comment by kartben** [comment]
I should add this is on macOS/Python 3.12 but can also reproduce on 3.10 on Linux

**Comment by dottspina** [comment]
Thank you for reporting this.

I was finally able to reproduce the issue ... by upgrading the Textualize [rich](https://github.com/Textualize/rich) library.

It seems that the problem in DTSh is rooted in a recent change to `rich.console` ([commit #a3b27a3](https://github.com/Textualize/rich/commit/a3b27a3)), which doesn't mean the bug is on their side: I'll dig into that.

In the meantime, could you please try downgrading the library (anything bellow version 13.8.0), e.g.:

```
pip install rich==13.7.1
```

This workaround should at least let you play with DTSh.

Thanks.

References (rich):
- [[BUG] Recorded content not expected for captured prints #2563](https://github.com/Textualize/rich/issues/2563)
- [Fix record and capture #3470](https://github.com/Textualize/rich/pull/3470)

**Comment by dottspina** [comment]
Automatically closed by commit ea1eaf5be738a41a3fb0789c98aa787170f56ceb: reopened to allow the issuer to confirm the fix.
Thanks.

**Comment by kartben** [comment]
This is now fixed indeed, thank you!
